bob-apps/bin/blob_recovery_tool/utils.rs

```diff
@@ -185,13 +244,16 @@ where
     info!("Input blob header version: {}", header.version);
     let mut count = 0;
     while !reader.is_eof() {
-        match reader.read_record_with_skip_wrong() {
+        match reader
+            .read_record_with_skip_wrong()
```

Comment:
This function is also used for migration, in which case it should not skip invalid entries. Migration must explicitly fail for broken BLOBs

Reply:
Fixed. Also I made skipping optional via flag `--skip-wrong-record`